builder/job.go

```diff
@@ -45,6 +45,16 @@ var validCommitCommands = map[string]bool{
 	"workdir":    true,
 }
 
+// BuiltinAllowedBuildArgs is list of built-in allowed build args
+var BuiltinAllowedBuildArgs = map[string]bool{
+	"HTTP_PROXY":  true,
```

Comment:
Yes, that is correct, 'true' is just a dummy value as I wanted a map for ease of search. I will make it a string with a default description. 

But note that since now we don't save these args in image metadata, so they can't be queried using `docker inspect`. But a third party tool that vendors in `builder` package will get access to these. Hope that is fine.